### PR TITLE
Consume icons from @enterprisedb/icons

### LINKED ISSUE #1534: Make icons private again?
Icons used to be installed from a Git submodule of a private repo.

Recently, we got rid of that private repo and moved the icons into this repository, making the icons public as a consequence.

If we have to make the icons private again I can see 2 potential options:

1. Go back to the Git submodule approach with the difference that now the repository will be hosted by the EnterpriseDB Github Org. (I'll see this as a temporary step before we get to number 2)
2. Move the icons to an internal npm package. We briefly talked about this before, there's already some infrastructure that will facilitate this but it will require some code refactoring.

### PR DESCRIPTION
**Pull request by frago12**
## What Changed?
* Consume icons from `@enterprisedb/icons` package
* Setup `direnv` to handle personal npm token so that users can install icons package
* Added dummy icons package to the repo for non-EDB Org members on Github
* Use `npm` instead of `yarn`

Closes #1534, #910

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+layout node
+source ./.envrc.local || echo "No .envrc.local found.  Create this file and export NPM_TOKEN to pull private dependencies."
```
Comment (gsong) [inline .envrc:2]: [suggest] https://direnv.net/man/direnv-stdlib.1.html#codesourceenvifexists-ltfilenamegtcode

File: .gitignore [hunk 2]
```diff
@@ -81,4 +78,8 @@ product_docs/content/
 product_docs/content_build/
 static/nginx_redirects.generated
 temp_kubernetes/
-temp_bdr/
\ No newline at end of file
+temp_bdr/
+
+# Track base direnv file
+!.envrc
+!.envrc
```
Comment (gsong) [inline .gitignore:85]: [?] Why twice?

File: README.md [hunk 3]
```diff
@@ -36,13 +36,33 @@ We recommend using MacOS to work with the EDB Docs application.
 
 1. Install Python 3 with `brew install python3`, if it's not already installed. (Use `python3 -V` to check that you have version 3.8 or higher.) Python is not needed for the core Gatsby system, but is required by several source scripts.
 
-1. Install Yarn with `npm i -g yarn`. Yarn is the package manager we're using for this project, instead of NPM. NPM may fail with a permissions related issue. To fix that, ensure that your user account owns the required directory: `sudo chown -R $(whoami) /usr/local/lib/node_modules`
+1. NPM is the package manager we're using for this project. NPM may fail with a permissions related issue. To fix that, ensure that your user account owns the required directory: `sudo chown -R $(whoami) /usr/local/lib/node_modules`
 
 1. Install Gatsby with `npm i -g gatsby-cli`. Gatsby is the software that powers the EDB Docs site.
 
-1. Install all required packages by running `yarn`.
+1. If you're a member of the EnterpriseDB Github Org, you'll want to follow these steps so that they're pulled in the next steps.
```
Comment (gsong) [inline README.md:43]: [suggest] Coordinate with @jaek-s on simplifying these steps, potentially. We should be able to have streamline directions for both OSS and internal contributors.

**Review by gsong — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -36,13 +36,33 @@ We recommend using MacOS to work with the EDB Docs application.
...
+
+   1. Create a Github token which can pull private packages. [You can learn how to do that here.](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#authenticating-to-github-packages)
+
+   1. Once you have that token, create a file in the root of the project called `.envrc.local` and add this code to it `export NPM_TOKEN="your-token-here"`.  That will be utilized in the NPM_TOKEN variable in the .npmrc file.
+
+1. If you're **not** a member of the EnterpriseDB Github Org, you'll have to follow these steps:
+
+  1. In the terminal, navigate to `icons-pkg/`, located folder in the root of the repository: `cd icons-pkg/`.
+
+  1. Link the icons package to you global npm packages: `npm link`.
```
Comment (josh-heyer) [inline README.md:55]: This (and the corresponding install step) strikes me as potentially error-prone - particularly for folks who have a working setup today and will need to go back & do stuff to keep it that way; can we capture this whole process in a script or something?

**Review by josh-heyer — COMMENTED** [review]
Looks fine; I need to set aside some time to test it (and note any gotchas for folks pulling down changes onto an existing dev environment), but don't expect any major issues.

Before this gets merged, we need to put together some guidance for the Doc team and others who may already be familiar with using Yarn; a cheat-sheet, etc. I'm a bit concerned that this will leave writers who are currently accustomed to the current workflow stranded until we get them set up. @jericson may have some ideas here, as we kinda just went through this with LFS :)

File: README.md (hunk 4 above)
Comment (frago12) [reply README.md:55]: @josh-heyer Jake is working on Dockerizing Docs to make things easier for the not expert users and I think that's going to help with this specifically. Stay tuned.

**Comment by gsong** [comment]
⚠️ Rebased!

**Comment by gsong** [comment]
⚠️ Force pushed, please fetch and reset accordingly.

File: docker/docker-compose.quickstart.yaml [hunk 5]
```diff
@@ -1,18 +1,21 @@
...
+    image: edb/docs-dev-server
+    container_name: docs-dev-server
+    hostname: docs-dev-server
     env_file:
       - ../.env.development
     init: true
+    working_dir: /app
+    command: >
+      sh -c "npm run install-dependencies &&
+      npm run develop -- -H 0.0.0.0 -p ${PORT:-8000} 2>&1"
```
Comment (gsong) [inline docker/docker-compose.quickstart.yaml:17]: @jaek-s FYI.

**Pull request merged**